### ISSUE TITLE
Cannot upload file to OpenVMS

### ISSUE THREAD

**Issue by sapr9x3** [robinrodricks/FluentFTP#380]
**FTP OS:** OpenVMS 8.4
**FTP Server:** OpenVMS
**Computer OS:** Windows 7 Professional SP1
**FluenFTP Nuget Version:** 23.0.0

I try to put a file to OpenVMS server and get the exception:

> Error while uploading the file to the server. See InnerException for more info. 

Inner exception:

> file specification syntax error

Inner stack trace:

>    в FluentFTP.FtpClient.OpenPassiveDataStream(FtpDataConnectionType type, String command, Int64 restart)
>   в FluentFTP.FtpClient.OpenDataStream(String command, Int64 restart)
>   в FluentFTP.FtpClient.GetNameListing(String path)
>   в FluentFTP.FtpClient.FileExists(String path)
>   в FluentFTP.FtpClient.UploadFileInternal(Stream fileData, String remotePath, Boolean createRemoteDir, > FtpExists existsMode, Boolean fileExists, Boolean fileExistsKnown, IProgress`1 progress)
My Code

External stack trace:

>   в FluentFTP.FtpClient.UploadFileInternal(Stream fileData, String remotePath, Boolean createRemoteDir, FtpExists existsMode, Boolean fileExists, Boolean fileExistsKnown, IProgress`1 progress)
>   в FluentFTP.FtpClient.UploadFileFromFile(String localPath, String remotePath, Boolean createRemoteDir, FtpExists existsMode, Boolean fileExists, Boolean fileExistsKnown, FtpVerify verifyOptions, IProgress`1 progress)
>   в FluentFTP.FtpClient.UploadFile(String localPath, String remotePath, FtpExists existsMode, Boolean createRemoteDir, FtpVerify verifyOptions, IProgress`1 progress)
>   в FluentFTPConsole.Program.Main(String[] args) в C:\VSProjs\PascalCodeEditor\FluentFTP\FluentFTPConsole\Program.cs:строка 71

My Code:

    string remoteDir = "sys$login";
    string filename = "test1.pas";
    var ftpclient = new FtpClient(Host, Credential);
    ftpclient.Connect();
    var tmpPath = Path.GetTempFileName();
    File.WriteAllText(tmpPath, content, encoding);
    ftpclient.SetWorkingDirectory(remoteDir);
    ftpclient.UploadDataType = FtpDataType.ASCII;			
    ftpclient.UploadFile(tmpPath, filename);

**Logs :**
```

# Connect()
Status:   Connecting to ***:21
Response: 220 mock FTP Server (Version 5.7) Ready.
Command:  USER ***
Response: 331 Username mylogin requires a Password
Command:  PASS ***
Response: 230 User logged in.
Command:  FEAT
Response: 211 END.
Response: 211-Extensions supported.
Response: AUTH TLS.
Response: PBSZ.
Response: PROT.
Status:   Text encoding: System.Text.ASCIIEncoding
Command:  SYST
Response: 200 VMS OpenVMS V8.4 on node mock.
Status:   Detected FTP server: OpenVMS
Status:   Auto-detected VMS listing parser

# SetWorkingDirectory("sys$login")
Command:  CWD sys$login
Response: 250 New default directory is SYS$SYSDEVICE:[USERS.mylogin]
Response: 250-CWD command successful.

# UploadFile("C:\Users\mylogin\AppData\Local\Temp\tmpB6BB.tmp", "test1.pas", Overw
rite, False, None)

# FileExists("test1.pas")

# GetWorkingDirectory()
Command:  PWD
Response: 257 "SYS$SYSDEVICE:[USERS.mylogin]" is current directory.

# GetNameListing("SYS$SYSDEVICE:[USERS.mylogin]")

# GetWorkingDirectory()
Command:  PWD
Response: 257 "SYS$SYSDEVICE:[USERS.mylogin]" is current directory.
Command:  TYPE I
Response: 200 TYPE set to IMAGE.

# OpenPassiveDataStream(AutoPassive, "NLST SYS$SYSDEVICE:[USERS.mylogin]/SYS$SYSDE
VICE:[USERS.mylogin]", 0)
Command:  EPSV
Response: 229 Entering Extended Passive Mode (|||50794|)
Status:   Connecting to ***:50794
Command:  NLST SYS$SYSDEVICE:[USERS.mylogin]/SYS$SYSDEVICE:[USERS.mylogin]
Response: 550 file specification syntax error
Status:   Disposing FtpSocketStream...

# CloseDataStream()

```

**Comment by robinrodricks** [comment]
Please try this and see if it solves?

[Release.zip](https://github.com/robinrodricks/FluentFTP/files/2914312/Release.zip)

**Comment by sapr9x3** [comment]
Yes. It works.



```
# Connect()
Status:   Connecting to ***:21
Response: 220 mock FTP Server (Version 5.7) Ready.
Command:  USER ***
Response: 331 Username mylogin requires a Password
Command:  PASS ***
Response: 230 User logged in.
Command:  FEAT
Response: 211 END.
Response: 211-Extensions supported.
Response: AUTH TLS.
Response: PBSZ.
Response: PROT.
Status:   Text encoding: System.Text.ASCIIEncoding
Command:  SYST
Response: 200 VMS OpenVMS V8.4 on node mock.
Status:   Detected FTP server: OpenVMS
Status:   Auto-detected VMS listing parser

# SetWorkingDirectory("sys$login")
Command:  CWD sys$login
Response: 250 New default directory is SYS$SYSDEVICE:[USERS.mylogin]
Response: 250-CWD command successful.

# UploadFile("C:\Users\mylogin\AppData\Local\Temp\tmp4151.tmp", "test1.pas", Overw
rite, False, None)

# FileExists("test1.pas")

# GetWorkingDirectory()
Command:  PWD
Response: 257 "SYS$SYSDEVICE:[USERS.mylogin]" is current directory.

# GetNameListing("SYS$SYSDEVICE:[USERS.mylogin]")
Command:  TYPE I
Response: 200 TYPE set to IMAGE.

# OpenPassiveDataStream(AutoPassive, "NLST SYS$SYSDEVICE:[USERS.mylogin]", 0)
Command:  EPSV
Response: 229 Entering Extended Passive Mode (|||50933|)
Status:   Connecting to ***:50933
Command:  NLST SYS$SYSDEVICE:[USERS.mylogin]
Response: 150 Opening data connection for SYS$SYSDEVICE:[USERS.mylogin] (192.168.1
60.102,55348)
Status:   Disposing FtpSocketStream...

# CloseDataStream()
Response: 226 NLST Directory transfer complete
Status:   Disposing FtpSocketStream...

# OpenWrite("test1.pas", ASCII)
Command:  TYPE A
Response: 200 TYPE set to ASCII.

# OpenPassiveDataStream(AutoPassive, "STOR test1.pas", 0)
Command:  EPSV
Response: 229 Entering Extended Passive Mode (|||50934|)
Status:   Connecting to ***:50934
Command:  STOR test1.pas
Response: 150 Opening data connection for SYS$SYSDEVICE:[USERS.mylogin]test1.pas;1
5 (192.168.160.102,55349)
Status:   Disposing FtpSocketStream...
Response: 226 Transfer complete.

```

**Comment by robinrodricks** [comment]
Released in [FluentFTP 23.1.0](https://www.nuget.org/packages/FluentFTP/23.1.0). Go grab it from nuget!
